pycytominer/cyto_utils/load.py

```diff
@@ -87,16 +148,40 @@ def load_profiles(profiles):
     FileNotFoundError
         Raised if the provided profile does not exists
     """
-    if not isinstance(profiles, pd.DataFrame):
-        # Check if path exists and load depending on file type
-        if is_path_a_parquet_file(profiles):
-            return pd.read_parquet(profiles, engine="pyarrow")
 
+    # If already a dataframe, return it
+    if isinstance(profiles, pd.DataFrame):
+        return profiles
+
+    # Check if path exists and load depending on file type
+    if is_path_a_parquet_file(profiles):
+        return pd.read_parquet(profiles, engine="pyarrow")
+
+    # Check if path is an AnnData file
+    anndata, anndata_type = is_anndata(profiles)
+    if anndata:
+        if anndata_type == "h5ad":
+            adata = ad.read_h5ad(profiles, backed="r")
+        elif anndata_type == "zarr":
+            adata = ad.read_zarr(profiles)
+        elif anndata_type == "in-memory":
+            adata = profiles
+        else:
+            raise ValueError("Unrecognized AnnData type")
```

Comment:
would we ever reach this line?

Reply:
it’s good to code defensively, but it should be an `AssertionError`, as we define the possible outputs.